### PR TITLE
Added bulk upload to ADLS

### PR DESCRIPTION
**Pull request by Diego-H-S**
## Summary
Here it is added to `task_utils.py` the task `adls_bulk_upload` used in the flows of Genesys and Mindful, but it can be also used in other flows when it is needed.

## Importance
It is an improvement of the code not relevant for business but clarifies the code and makes it easy to read and use.

## Checklist

This PR:

- [x] adds new tests (if appropriate)
- [x] updates docstrings for any new functions or function arguments (if appropriate)
- [x] updates `CHANGELOG.md` with a summary of the changes

### CODE REVIEW

File: viadot/task_utils.py [hunk 1]
```diff
@@ -570,3 +570,36 @@ def credentials_loader(credentials_secret: str, vault_name: str = None) -> dict:
...
+    Args:
+        file_names (List[str]): List of file names to generate paths.
+        file_name_relative_path (str, optional): Path where to save the file locally. Defaults to ''.
+        adls_file_path (str, optional): Azure Data Lake path. Defaults to None.
+        adls_sp_credentials_secret (str, optional): The name of the Azure Key Vault secret containing a dictionary with
+            ACCOUNT_NAME and Service Principal credentials (TENANT_ID, CLIENT_ID, CLIENT_SECRET). Defaults to None.
+        adls_overwrite (bool, optional): Whether to overwrite files in the data lake. Defaults to True.
+
+    Returns:
+        List[str]: List of paths
```
Comment (Rafalz13) [inline viadot/task_utils.py:594]: Add full stop.

**Review by Rafalz13 — CHANGES_REQUESTED** [review]
Looks good.
Add tests for new function.

File: viadot/task_utils.py (hunk 1 above)
Comment (Diego-H-S) [reply viadot/task_utils.py]: Added ;)

**Review by Rafalz13 — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,7 +5,13 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+### Added
+- Added `adls_bulk_upload` task function to `task_utils.py`
+- Added `test_adls_bulk_upload` test function to `test_task_utils.py`
```
Comment (Rafalz13) [inline CHANGELOG.md:10]: Information about tests is not important for the changelog. We are storing only information about the viadot code.

File: tests/unit/test_task_utils.py [hunk 3]
```diff
@@ -18,9 +19,20 @@
     dtypes_to_json_task,
     union_dfs_task,
     write_to_json,
+    adls_bulk_upload,
 )
 
 
+class MockClass:
```
Comment (Rafalz13) [inline tests/unit/test_task_utils.py:26]: Is this class only for `adls_bulk_upload` or can it be used for other tests? You have specific arguments there, so maybe change the name of this class to be easier to read and use.

**Review by Rafalz13 — COMMENTED** [review]
Looks good. 2 changes to consider.

File: CHANGELOG.md (hunk 2 above)
Comment (Diego-H-S) [reply CHANGELOG.md:10]: removed!

File: tests/unit/test_task_utils.py (hunk 3 above)
Comment (Diego-H-S) [reply tests/unit/test_task_utils.py]: done!

**Pull request merged**